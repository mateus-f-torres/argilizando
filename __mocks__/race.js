import RACES from '../src/data/races.js';

const getRace = (raceID) => {
  for (const category of RACES) {
    for (const item of category[1]) {
      if (item.id === raceID) return item;
    }
  }
};

export const mockHuman = Object.assign({}, getRace('human'));

export const mockOrc = Object.assign({}, getRace('orc'));
