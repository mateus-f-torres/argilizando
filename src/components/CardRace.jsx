//@flow
import * as React from 'react';
import {Link} from 'react-router-dom';
import type {Race} from '../types/props.js';

import '../data/raceImages.js';

type Props = Race & {
  lockRace: () => void
};

const CardRace = (props: Props) => {
  return (
    <figure className="card">
      <h2>{props.name}</h2>
      <img className="race-img" src={`./../images/${props.id}.png`} />
      <div className="horizontal-line" />
      <ul id="race-stats" className="stats">
        <h3>Race Stats</h3>
        <li><p>
          <b>Height:</b>{props.height}.
        </p></li>
        <li><p>
          <b>Weight:</b>{props.weight}.
        </p></li>
        <li><p>
          <b>Adulthood:</b>{props.age[0]} years.
        </p></li>
        <li><p>
          <b>Lifespan:</b>{props.age[1]} years.
        </p></li>
        <li><p>
          <b>Size:</b>{props.size}.
        </p></li>
        <li><p>
          <b>Speed:</b>{props.speed}ft.
        </p></li>
        <li><p>
          <b>Speaks:</b>{`${props.lang[0]} & ${props.lang[1]}.`}
        </p></li>
      </ul>
      <div className="horizontal-line" />
      <ul id="race-traits">
        <h3>Race Traits</h3>
        {
          props.special.map((trait, i) => (
            <li key={i}>
              <h4>{trait[0]} :</h4>
              <p>{trait[1]}</p>
            </li>
          ))
        }
      </ul>
      <div className="horizontal-line no-bottom" />
      <div className="lock-box">
        <Link to="/race">
          <button>Back</button>
        </Link>
        <Link to="/">
          <button
            data-test="race-lock"
            onClick={props.lockRace}>Lock</button>
        </Link>
      </div>
    </figure>
  );
};

export default CardRace;
