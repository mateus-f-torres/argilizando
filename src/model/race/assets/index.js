import human from './human.json'

export default {
  human,
}