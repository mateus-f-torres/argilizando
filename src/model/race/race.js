import RACES from './assets'

/*
Common Races
[common races flavor text]
  - Human
    [brief human flavor text]
  - Halfling
    [brief halfling flavor text]

Uncommon Races
[uncommon races flavor text]

Monstrous Races
[monstrous races flavor text]
 */
export function getAllRacesDescription() {}

/*
[Race], [Race Category]

[extended race flavor text]

"age"
"size"
"height"
"weight"
"eyes"
"skin"
"hair"
"speed"
"language"
"special_traits": []
 */
export function getDetailedRaceDescription(raceId) {
  return RACES.human
}

/*
[race mechanical stats]
id
speed
language* (some races get to choose a second language later)
traits** (some races alter other game stats)
 */
export function getChoosenRace(raceId) {}
