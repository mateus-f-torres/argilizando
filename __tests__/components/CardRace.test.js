import React from 'react';
import {mount} from 'enzyme';
import {MemoryRouter} from 'react-router-dom';
import CardRace from 'Components/CardRace.jsx';
import {mockHuman} from 'Mock/race.js';

describe('<CardRace />', () => {
  test('call lockRace when lock button clicked', () => {
    const mockOnClick = jest.fn();
    const wrapper = mount(
      <MemoryRouter initialEntries={
        [
          {pathname: '/race/human', key: 'testKey'},
        ]
      }>
        <CardRace lockRace={mockOnClick} {...mockHuman} />
      </MemoryRouter>
    );

    wrapper.find('[data-test="race-lock"]').simulate('click');

    expect(mockOnClick.mock.calls.length).toBe(1);
  });

  test('CardRace snapshot matches current production CardRace', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={
        [
          {pathname: '/race/human', key: 'testKey'},
        ]
      }>
        <CardRace {...mockHuman} />
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
