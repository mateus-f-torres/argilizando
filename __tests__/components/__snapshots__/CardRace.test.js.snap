// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardRace /> match current production CardRace 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/race/human",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/race/human",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/race/human",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CardRace
      _placeholder="Salazar Marivaldi"
      age={
        Array [
          20,
          80,
        ]
      }
      eyes="All human variations"
      hair="All human variations"
      height="6ft (1,7m)"
      id="human"
      lang={
        Array [
          "Common",
          "..",
        ]
      }
      name="Human"
      size="Medium"
      skin="All human variations"
      special={
        Array [
          Array [
            "Bonus Skill",
            "You gain proficiency in one additional skill 
            from your class skill list.",
          ],
          Array [
            "Magical Affinity",
            "Whenever you are 5ft from a magical item you can sense 
            its magic through one of your senses. 
            Once per short rest you can increase the range of this feature 
            to 30ft for 1min, after that you lose this feature and 
            must finish a short rest to regain its passive effect. 
            Using this feature only tells you if something is magical or not, 
            nothing more than that.",
          ],
          Array [
            "Grit",
            "You can use your reaction to gain advantage on a 
            saving throw against a spell. 
            You must roll a 5 or 6 on a d6 to use this feature again.",
          ],
        ]
      }
      speed={30}
      weight="180lb (80kg)"
    >
      <figure
        className="card"
      >
        <h2>
          Human
        </h2>
        <img
          className="race-img"
          src="/test-file-stub"
        />
        <div
          className="horizontal-line"
        />
        <ul
          className="stats"
          id="race-stats"
        >
          <h3>
            Race Stats
          </h3>
          <li>
            <p>
              <b>
                Height:
              </b>
              6ft (1,7m)
              .
            </p>
          </li>
          <li>
            <p>
              <b>
                Weight:
              </b>
              180lb (80kg)
              .
            </p>
          </li>
          <li>
            <p>
              <b>
                Adulthood:
              </b>
              20
               years.
            </p>
          </li>
          <li>
            <p>
              <b>
                Lifespan:
              </b>
              80
               years.
            </p>
          </li>
          <li>
            <p>
              <b>
                Size:
              </b>
              Medium
              .
            </p>
          </li>
          <li>
            <p>
              <b>
                Speed:
              </b>
              30
              ft.
            </p>
          </li>
          <li>
            <p>
              <b>
                Speaks:
              </b>
              Common & ...
            </p>
          </li>
        </ul>
        <div
          className="horizontal-line"
        />
        <ul
          id="race-traits"
        >
          <h3>
            Race Traits
          </h3>
          <li
            key="0"
          >
            <h4>
              Bonus Skill
               :
            </h4>
            <p>
              You gain proficiency in one additional skill 
            from your class skill list.
            </p>
          </li>
          <li
            key="1"
          >
            <h4>
              Magical Affinity
               :
            </h4>
            <p>
              Whenever you are 5ft from a magical item you can sense 
            its magic through one of your senses. 
            Once per short rest you can increase the range of this feature 
            to 30ft for 1min, after that you lose this feature and 
            must finish a short rest to regain its passive effect. 
            Using this feature only tells you if something is magical or not, 
            nothing more than that.
            </p>
          </li>
          <li
            key="2"
          >
            <h4>
              Grit
               :
            </h4>
            <p>
              You can use your reaction to gain advantage on a 
            saving throw against a spell. 
            You must roll a 5 or 6 on a d6 to use this feature again.
            </p>
          </li>
        </ul>
        <div
          className="horizontal-line no-bottom"
        />
        <div
          className="lock-box"
        >
          <Link
            replace={false}
            to="/race"
          >
            <a
              href="/race"
              onClick={[Function]}
            >
              <button>
                Back
              </button>
            </a>
          </Link>
          <Link
            replace={false}
            to="/"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <button
                data-test="race-lock"
              >
                Lock
              </button>
            </a>
          </Link>
        </div>
      </figure>
    </CardRace>
  </Router>
</MemoryRouter>
`;
